### PR TITLE
enhancement: Add query planner filter kind

### PR DESCRIPTION
**Pull request by dbuduev**
#### Description

Convert `response.Filter` to an object in order to improve the readability of a response for always allowed/denied cases.

#### Checklist 

- [x] The PR title has the correct prefix 
- [ ] PR is linked to the corresponding issue
- [x] All commits are signed-off (`git commit -s ...`) to provide the [DCO](https://developercertificate.org/)

### CODE REVIEW

File: api/public/cerbos/response/v1/response.proto [hunk 1]
```diff
@@ -72,11 +72,29 @@ message ResourcesQueryPlanResponse {
...
+           KIND_CONDITIONAL = 3;
+        }
+        Kind kind = 1 [
+            (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+                description: "Filter kind",
+            }
+        ];
+        Expression.Operand condition = 2 [
+            (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+                description: "Filter condition",
```
Comment (charithe) [inline api/public/cerbos/response/v1/response.proto:89]:
```suggestion
                description: "Filter condition. Only populated if kind is KIND_CONDITIONAL.",
```

File: api/public/cerbos/response/v1/response.proto [hunk 2]
```diff
@@ -72,11 +72,29 @@ message ResourcesQueryPlanResponse {
...
+    message Filter {
+        enum Kind {
+           KIND_UNSPECIFIED = 0;
+           KIND_ALWAYS_ALLOWED = 1;
+           KIND_ALWAYS_DENIED = 2;
+           KIND_CONDITIONAL = 3;
+        }
+        Kind kind = 1 [
+            (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+                description: "Filter kind",
```
Comment (charithe) [inline api/public/cerbos/response/v1/response.proto:84]:
```suggestion
                description: "Filter kind. Defines whether the given action is always allowed, always denied or allowed conditionally.",
```

**Review by charithe — COMMENTED** [review]
LGTM. I think the documentation page needs to be updated to reflect the new shape of the request and explain what `filter.kind` is.

**Comment by charithe** [comment]
I created #559 to document the response format. Are there any examples or tests that use the operators listed in #560?

**Comment by dbuduev** [comment]
> Are there any examples or tests that use the operators listed in #560

Only unit tests. test file  - `ast_test.go` in the package `engine`; testdata - ast_build_expr.yaml.

**Review by charithe — DISMISSED** [review]
LGTM. Will need to run `make generate` to update the Swagger definitions though.

**Review by charithe — APPROVED** [review]
LGTM.

I removed the "breaking" modifier from the title because this feature is not yet released and therefore it's not technically breaking anything.

**Pull request merged**